fold/dna.go

```diff
@@ -0,0 +1,607 @@
+package fold
+
+var DNA_COMPLEMENT = map[byte]byte{'A': 'T', 'T': 'A', 'G': 'C', 'C': 'G', 'N': 'N'}
+
+var DNA_MULTIBRANCH = MultiBranchEnergies{A: 2.6, B: 0.2, C: 0.2, D: 2.0}
+
+// The Thermodynamics of DNA Structural Motifs
+// SantaLucia and Hicks, 2004
+var DNA_NN = BPEnergy{"AA/TT": Energy{H: -7.6, S: -21.3},
+	"AC/TG":    {H: -8.4, S: -22.4},
+	"AG/TC":    {H: -7.8, S: -21},
+	"AT/TA":    {H: -7.2, S: -20.4},
+	"C/G_tini": {H: 0, S: 0},
+	"CA/GT":    {H: -8.5, S: -22.7},
+	"CC/GG":    {H: -8, S: -19.9},
+	"CG/GC":    {H: -10.6, S: -27.2},
+	"CT/GA":    {H: -7.8, S: -21},
+	"GA/CT":    {H: -8.2, S: -22.2},
+	"GC/CG":    {H: -9.8, S: -24.4},
+	"GG/CC":    {H: -8, S: -19.9},
+	"GT/CA":    {H: -8.4, S: -22.4},
+	"T/A_tini": {H: 2.2, S: 6.9},
+	"TA/AT":    {H: -7.2, S: -21.3},
+	"TC/AG":    {H: -8.2, S: -22.2},
+	"TG/AC":    {H: -8.5, S: -22.7},
+	"TT/AA":    {H: -7.6, S: -21.3},
+	"init":     {H: 0.2, S: -5.7},
+	"init_A/T": {H: 2.2, S: 6.9},
+	"init_G/C": {H: 0, S: 0},
+	"mys":      {H: 0, S: -1.4},
+	"sym":      {H: 0, S: -1.4},
+	"tini":     {H: 0.2, S: -5.7}}
+
+// Internal mismatch table (DNA)
+// Allawi & SantaLucia (1997), Biochemistry 36: 10581-10594
+// Allawi & SantaLucia (1998), Biochemistry 37: 9435-9444
+// Allawi & SantaLucia (1998), Biochemistry 37: 2170-2179 *
+// Allawi & SantaLucia (1998), Nucl Acids Res 26: 2694-2701 *
+// Peyret et al. (1999), Biochemistry 38: 3468-3477 *
+var DNA_INTERNAL_MM = BPEnergy{"AA/AT": Energy{H: 4.7, S: 12.9},
```

Comment:
trying to change fold.FoldEnergy to fold.Energy but everytime I do I get a compiler error here:

````
initialization cycle for DNA_INTERNAL_MM
	./dna.go:40:5: DNA_INTERNAL_MM refers to
	./fold.go:45:6: Energy refers to
	./fold.go:28:6: Fold refers to
	./seqfold.go:128:6: NewFoldingContext refers to
	./dna.go:596:5: DNA_ENERGIES refers to
	./dna.go:40:5: DNA_INTERNAL_MMgo
  ```